pkg/resource/deploy/source_eval.go

```diff
@@ -386,6 +396,42 @@ func (d *defaultProviders) handleRequest(req providers.ProviderRequest) (provide
 	return ref, nil
 }
 
+// If req should be allowed, or if we should prevent the request.
+func (d *defaultProviders) shouldDenyRequest(req providers.ProviderRequest) (bool, error) {
+	logging.V(9).Infof("checking if %s should be denied", req)
+	pConfig, err := d.config.GetPackageConfig("pulumi")
+	if err != nil {
+		return true, err
+	}
+
+	denyCreation := false
+	if value, ok := pConfig["disable-default-providers"]; ok {
+		array := []interface{}{}
+		if !value.IsString() {
+			return true, fmt.Errorf("Unexpected endecoding of pulumi:disable-default-providers")
+		}
+		if err := json.Unmarshal([]byte(value.StringValue()), &array); err != nil {
```

Comment:
Our config is YAML why don't we allow proper array options? String will be fine for now but it would be a nice improvement to config.

Reply:
The config (`Pulumi.*.yaml`) actually has a proper array. We serialize it to json as some earlier step. I'm not sure why it isn't kept an array.